src/main/java/codeu/model/data/Event.java

```diff
@@ -0,0 +1,173 @@
+
+package codeu.model.data;
+
+import java.time.Instant;
+import java.util.UUID;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.util.Locale;
+import java.time.ZoneId; 
+
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+
+/** 
+ *  This class abstracts any object so it can be a User, Conversation, or Message. 
+ *  Made so a list can  easily hold and sort all types of objects at once.
+ */
+public class Event{
+
+	/* Denotes the type of event this is. 'c' for conversation, 'm' for message, and 'u' for user. */
+	private final Character eventType;
+
+	/* All the fields needed from the conversation class. */
+	private final String titleOfConversation;
+	private final UUID authorIdForConversation;
+	private final Instant conversationCreationTime;
+
+	/* All the fields needed from the message class. */
+	private final Instant messageCreationTime;
+    private final UUID authorIdForMessage;
+    private final String conversationTitleOfMessage;
+    private final String messageContent;
+
+    /* All the fields needed from the user class. */
+    private final Instant userCreationTime;
+    private final String nameOfUser;
+
+    /* Needed to get the user given the ID from a conversation object. */
+    private final UserStore userStore;
+    /* Need to get the conversation given the ID from a message object. */
+    private final ConversationStore conversationStore;
+  
+  /**
+   * Constructs a new event using a Conversation object.
+   *
+   * @param conversation an object of type Conversation
+   */
+	public Event(Conversation conversation) {
+      userStore = UserStore.getInstance();
+      conversationStore = ConversationStore.getInstance();
```

Comment:
Again, I understand your approach of using the notion of Event for your sorting purpose.
You can proceed with this change.
But I want to talk to you at some point about the design drawback of this approach. How code and data need to be duplicated to support this, which is undesirable.